dev/PushNotifications/PushNotificationManager.cpp

```diff
@@ -245,4 +245,36 @@ namespace winrt::Microsoft::Windows::PushNotifications::implementation
             LOG_IF_FAILED(::CoRevokeClassObject(static_cast<DWORD>(token.Cookie())));
         }
     }
+
+    bool PushNotificationManager::IsBIAvailable()
+    {
+        try
+        {
+            BackgroundTaskBuilder builder = BackgroundTaskBuilder();
+            auto builder5 = builder.as<winrt::IBackgroundTaskBuilder5>();
+        }
+        catch (...)
+        {
+            return false;
+        }
+        return true;
+    }
+
+    bool PushNotificationManager::IsActivatorSupported(PushNotificationRegistrationOptions const& options)
+    {
+        bool isBIFlagSet = WI_IsAnyFlagSet(options, PushNotificationRegistrationOptions::PushTrigger | PushNotificationRegistrationOptions::ComActivator);
+        bool isProtocolActivatorSet = WI_IsFlagSet(options, PushNotificationRegistrationOptions::ProtocolActivator);
+
+        THROW_HR_IF(E_INVALIDARG, isBIFlagSet && isProtocolActivatorSet); // Invalid flag combination
+        THROW_HR_IF(E_INVALIDARG, !(isBIFlagSet || isProtocolActivatorSet)); // null flags
```

Comment:
Hmm is this second state even possible? I thought the enum forced you to select atleast one activator type. If so, can we remove this second check?

Reply:
Someone could do
`PushNotificationRegistrationOptions option;
 IsActivatorSupported(option)`
And option would init as 0

Reply:
Then why not check for 0 in the beginning of the function?

Reply:
Even better, we could set Undefined to 0 and check for that in the beginning of the function.